src/realm/alloc_slab.cpp

```diff
@@ -818,8 +820,9 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
             realm::util::encryption_read_barrier(writable_map, 0);
             writable_header.m_flags |= flags_SelectBit;
             realm::util::encryption_write_barrier(writable_map, 0);
-            m_file_on_streaming_form = false;
             writable_map.sync();
+
+            realm::util::encryption_read_barrier(m_file_mappings->m_initial_mapping, 0, sizeof(Header));
```

Comment:
The call cannot be done at that point, because the allocator might not be attached to a file, but to a buffer.